### PR TITLE
fix: reuse one instance of Random, instead of creating new each time

### PR DESCRIPTION
**Pull request by stian-sandvold**
Suggestion by sonarCloud, see issue for info

### CODE REVIEW

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/MathUtils.java [hunk 1]
```diff
@@ -71,6 +71,8 @@
 
     private static final Pattern ZERO_PATTERN = Pattern.compile( "^0(\\.0*)?$" );
 
+    private static final Random RANDOM = new Random();
+
```
Comment (netroms) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/MathUtils.java:75]: Might be better to use a ThreadLocalRandom.
See Random.class java doc: Consider instead using java.util.concurrent.ThreadLocalRandom in multithreaded designs.

**Review by netroms — APPROVED** [review]

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/MathUtils.java (hunk 1 above)
Comment (stian-sandvold) [reply dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/MathUtils.java:75]: Thanks, I'll check it out!

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/MathUtils.java [hunk 2]
```diff
@@ -71,6 +72,8 @@
 
     private static final Pattern ZERO_PATTERN = Pattern.compile( "^0(\\.0*)?$" );
 
+    private static final Random RANDOM = ThreadLocalRandom.current();
+
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/MathUtils.java:76]: Just wondering if this is a great idea, as a class scope variable means that all threads will use the same instance.

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html

From the javadoc it seems the recommended usage is:

```
ThreadLocalRandom.current().nextX(...) 
```
I.e. `ThreadLocalRandom.current()` returns the current thread's ThreadLocalRandom.


[current](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html#current--)()
Returns the current thread's ThreadLocalRandom.

Comment (stian-sandvold) [reply dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/MathUtils.java:76]: I think you are right, I'll remedy this right away!

**Review by larshelge — APPROVED** [review]

**Pull request merged**